plutok-flat/src/main/kotlin/io/newm/plutok/uplc/ast/Program.kt

```diff
@@ -1,10 +1,32 @@
 package io.newm.plutok.uplc.ast
 
+import io.newm.plutok.uplc.flat.FlatEncoder
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
 /**
  * This represents a program in Untyped Plutus Core (uplc).
  * It contains a version and a term that requires a generic type.
  */
+@Serializable(with = ProgramSerializer::class)
 data class Program<T>(
     val version: Version,
     val term: Term<T>,
 )
+
+object ProgramSerializer : KSerializer<Program<*>> {
```

Comment:
Also if we only need to implement the serialize method we can extend `SerializationStrategy<T>` instead

Reply:
We'll need to implement both serialize and deserialize eventually.